### PR TITLE
MIAD-85 creation of new create alert endpoint

### PR DESCRIPTION
**Pull request by anthony-britton-moj**

### CODE REVIEW

**Comment by ReedSoftware** [comment]
@anthony-britton-moj I don't think you were around for the talk between Dave, Bill and myself about API testing - if the service tests you've modified seem redundant next to the int tests, do feel free to remove them. It does look like you've ended up changing the same thing twice in the Int test and service tests. If there's maybe 1 or 2 scenarios covered by the service tests that the Int tests don't cover, it'd still be worth (imo) to remove the service tests entirely, and just move those scenarios to the Int tests. Slice-of-behaviour testing over adhering exactly to tests at each layer is what we agreed made more sense to do, most of the time.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/PrisonerAlertsController.kt [hunk 1]
```diff
@@ -143,4 +149,62 @@ class PrisonerAlertsController(val alertService: AlertService) {
...
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @PostMapping("/{prisonNumber}/alerts")
+  @ResponseStatus(HttpStatus.CREATED)
+  @PreAuthorize("hasAnyRole('$ROLE_ALERTS_WRITER', '$ROLE_ALERTS_ADMIN', '$UPDATE_ALERT', '$ROLE_NOMIS_ALERTS')")
+  @UsernameHeader
+  @SourceHeader
+  fun createAlert(
```
Comment (iterative-it) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/PrisonerAlertsController.kt:190]: Tiny point here. The Swagger docs will use this in the URL and as there are two, it'll use createAlert and createAlert_1. I think there's another instance of it in the migrate alert controller. Maybe `createPrisonerAlert`?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/CreateAlertIntTest.kt [hunk 2]
```diff
@@ -158,7 +158,7 @@ class CreateAlertIntTest : IntegrationTestBase() {
       assertThat(status).isEqualTo(400)
       assertThat(errorCode).isNull()
       assertThat(userMessage).isEqualTo("Validation failure: Couldn't read request body")
-      assertThat(developerMessage).isEqualTo("Required request body is missing: public uk.gov.justice.digital.hmpps.hmppsalertsapi.model.Alert uk.gov.justice.digital.hmpps.hmppsalertsapi.resource.AlertsController.createAlert(uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.CreateAlert,jakarta.servlet.http.HttpServletRequest)")
+      assertThat(developerMessage).isEqualTo("Required request body is missing: public uk.gov.justice.digital.hmpps.hmppsalertsapi.model.Alert uk.gov.justice.digital.hmpps.hmppsalertsapi.resource.PrisonerAlertsController.createAlert(java.lang.String,uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.CreateAlert,jakarta.servlet.http.HttpServletRequest)")
```
Comment (iterative-it) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/CreateAlertIntTest.kt:161]: These tests have proven brittle and they don't really add much value. I changed some of them to `startsWith("Required request body is missing: ")`

**Review by iterative-it — APPROVED** [review]
LGTM

**Pull request merged**